pkgs/development/libraries/ncurses/default.nix

```diff
@@ -46,6 +46,9 @@ stdenv.mkDerivation rec {
   buildInputs = lib.optional (mouseSupport && stdenv.isLinux) gpm;
 
   preConfigure = ''
+    # These paths end up in the default lookup chain.
+    export TERMINFO_DIRS=/etc/terminfo
+
     export PKG_CONFIG_LIBDIR="$dev/lib/pkgconfig"
```

Comment:
Well, it is always nice to learn new things, and I really enjoy getting to the bottom of an issue/disagreement. You are indeed correct that the quotes are not needed here.
I think I can also understand your irritation at seeing code that does not match your standards. I also expect contributors to fix issues they encounter around their patches.

What I do not understand is why you consider extra quoting a big issue. It is common practice to quote bash strings, even when not needed. This shows that the content is intended to be a string. It is a bit like curly braces for blocks in C. If your block is only one statement, curly braces are not needed, but nobody ever asks you to remove them. While redundant, they are correct and convey intent. 
The same applies with export: `export CFLAGS="-Wall"` does not need quotes, but if you add another argument, then you will want the quotes, as in `export CFLAGS="-Wall -O1".

I will not change this, because it does not belong to this commit, and is not needed. You are of course free to fix it in another PR, but I urge you not to do that. While you really want to get rid of useless quotes, there will be someone else out there that insists on quoting every string. In the absence of a clear consensus, this will end up in a load of commits toggling quotes, with no added value. It will be even worse if these toggle happen together with unrelated changes.
For now, there is about 600 `export` statements with quotes in nixpkgs, and the same without quotes. No consensus.

I really want to make a point here, because I did not appreciate being called on this non error. Like everyone else, I try to make my PR as good as possible and I appreciate constructive comments and technical discussions. I however find that subjective comments about code formatting make submitting PRs a pain. I do not understand why you would want to lose time even noticing that, and I am afraid that you may make the same comments to other contributors. 
Can you understand how irritating it is to get these comments when members commit code like that all the time ? Even Eelco does that sometimes (3fe93d2f75b36a2663a377826fc792b8207c3f7b, 452afd1ed68a53e5c845f9df96d707b24d40b250, 81e530a7499e3f06bec13b6ec974d0f1e3a4ab84) even though not all the time, and sometimes alternating style in the same commit (497c828611f248baa8bdf8320e841b24e528bbe4). From his coding style, I am not even sure he knows about the above POSIX rules around export.

I understand that you may be annoyed, but it should no be at the expense of random contributors that just happen not to have direct commit access. If you want this to change, please make it a general discussion on the ML.